src/Database/Relations/MorphTo.php

```diff
@@ -22,6 +22,90 @@ public function __construct(Builder $query, Model $parent, $foreignKey, $otherKe
         $this->addDefinedConstraints();
     }
 
+    /**
+     * Associate the model instance to the given parent.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    public function associate($model)
+    {
+        /**
+         * @event model.relation.beforeAssociate
+         * Called before associating a relation to the model (only for BelongsTo/MorphTo relations)
+         *
+         * Example usage:
+         *
+         *     $model->bindEvent('model.relation.beforeAssociate', function (string $relationName, \October\Rain\Database\Model $relatedModel) use (\October\Rain\Database\Model $model) {
+         *         if ($relationName === 'dummyRelation') {
+         *             throw new \Exception("Invalid relation!");
+         *         }
+         *     });
+         *
+         */
+        $this->parent->fireEvent('model.relation.beforeAssociate', [$this->relationName, $model]);
+
+        $result = parent::associate($model);
+
+        /**
+         * @event model.relation.afterAssociate
+         * Called after associating a relation to the model (only for BelongsTo/MorphTo relations)
+         *
+         * Example usage:
+         *
+         *     $model->bindEvent('model.relation.afterAssociate', function (string $relationName, \October\Rain\Database\Model $relatedModel) use (\October\Rain\Database\Model $model) {
+         *         $relatedClass = get_class($relatedModel);
+         *         $modelClass = get_class($model);
+         *         traceLog("{$relatedClass} was associated as {$relationName} to {$modelClass}.");
+         *     });
+         *
+         */
+        $this->parent->fireEvent('model.relation.afterAssociate', [$this->relationName, $this->related]);
```

Comment:
Missed last one `$this->related`, really last :-)

Reply:
done!